File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0; int tot = 0, i;
        for(i = 0; i < nums.size(); i++){
            sum += nums[i];
            tot += i;
        }
        tot += i;
        return tot - sum;
    }
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] arr) {
       int i = 0;
       while(i < arr.length){
           int val = arr[i];
           if(val < arr.length && val != arr[val]) swap(arr, i, val);
           else i++;
       } 
       for(int j = 0; j < arr.length; j++){
           if(arr[j] != j) return j;
       }
       return arr.length;
    }
    public void swap(int[] arr, int a, int b){
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }
}